packages/contracts/contracts/SmartInvoiceEscrow.sol

```diff
@@ -206,6 +215,37 @@ contract SmartInvoiceEscrow is
         emit MetaEvidence(metaEvidenceId, initData.details);
     }
 
+    function _handleResolver(
+        address _resolver,
+        bytes memory _resolverData
+    ) internal {
+        if (_resolver == address(0)) revert InvalidResolver();
+        if (_resolverData.length < 32) revert InvalidResolverData();
+
+        uint8 _resolverType = abi.decode(_resolverData, (uint8));
+        if (_resolverType > uint8(ADR.ARBITRATOR)) revert InvalidResolverType();
+
+        if (_resolverType == uint8(ADR.INDIVIDUAL)) {
+            // Expect exactly one static word: abi.encode(uint8)
+            if (_resolverData.length != 32) revert InvalidResolverData();
+            uint256 _resolutionRateBPS = FACTORY.resolutionRateOf(_resolver);
+            if (_resolutionRateBPS > 1000) revert InvalidResolutionRate();
+            resolutionRateBPS = _resolutionRateBPS;
+        } else {
+            // Expect tuple encoding: abi.encode(uint8, bytes)
+            if (_resolverData.length < 96) revert InvalidResolverData();
```

Comment:
I don't think length of 96 is enough. I checked it on Remix and it seems that to decode `bytes` we'll need additional values for offset and length of bytes array. Decoding worked for me when I used this value which is 160 bytes
0x0000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000000000000000000018000000000000000000000000000000000000000000000000000000000000000f

Reply:
Sure. I'll take a closer look at this flow.

Reply:
fixed in latest commit